src/d3a/d3a_core/sim_results/bills.py

```diff
@@ -297,9 +262,6 @@ def _accumulate_market_fees(self, area, past_market_types):
             self._accumulate_market_fees(child, past_market_types)
 
     def _update_market_fees(self, area, market_type):
-        if ConstSettings.GeneralSettings.KEEP_PAST_MARKETS:
-            # If all the past markets remain in memory, reinitialize the market fees
-            self.market_fees = {}
         self._accumulate_market_fees(area, market_type)
 
     def update(self, area):
```

Comment:
Good time to think of solving some more memory issues via trying to getting rid of duplicate results dict for backend and redis like [here](https://github.com/gridsingularity/d3a/pull/839/files?file-filters%5B%5D=.py#diff-dbad1988984f5b467d3f69255b611e42R273-R274) in L273-L274 at this point of commits.

Reply:
True! Let me try that.

Reply:
@faizan2590 A gave my best, but it is not possible without changing the whole (very complicated) structure of the bills creation.
The problem is, that the bills structure is so different for areas and devices. The code would bemore complicated than the actual saving of memory. Regarding the memory: Even for large grids, the bills dont take up a lot of space, so i suggest to go with the easy solution and to only populate the bills_redis_results if `should_export_plots is False`. Does it make sense?

Reply:
Ok, i thought about it some more. the cumulative_bills are created with uuid keys, too. 
To be consistent, I would exchange all "name" handling in the MarketBills into uuids and only exchange them into area names just before exporting to local json. OK?

Reply:
OK, I gave it another try, but it is too complicated. The problem is, that the root keys in the bills dict are uuids, but all subdictkeys are area names. This is not possible to handle with the current bills implementation. Will go with the way proposed in my last comment @faizan2590

Reply:
Actually, I just reverted my changes as CumulativeBills is depending on MarketEnergyBills.bills_redis_results.